### ISSUE TITLE
Savefile does not work with latest rust nightly

### ISSUE THREAD

**Issue by avl** [avl/savefile#76]
The test-suite fails with lots of errors like:
```

error: non-local `impl` definition, `impl` blocks should be written at the same level as their item
  --> savefile-test/src/savefile_abi_test/enum_tests.rs:12:1
   |
12 | #[savefile_abi_exportable(version = 0)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | `__14_` is not local
   | `AbiExportable` is not local
   | move the `impl` block outside of this constant `_` and up 2 bodies
   |
   = note: the attribute macro `savefile_abi_exportable` defines the non-local `impl`, and may need to be changed
   = note: the attribute macro `savefile_abi_exportable` may come from an old version of the `savefile_derive` crate, try updating your dependency with `cargo update -p savefile_derive`
   = note: an `impl` is never scoped, even when it is nested inside an item, as it may impact type checking outside of that item, which can be the case if neither the trait or the self type are at the same nesting level as the `impl`
   = note: items in an anonymous const item (`const _: () = { ... }`) are treated as in the same scope as the anonymous const's declaration for the purpose of this lint
   = note: this lint may become deny-by-default in the edition 2024 and higher, see the tracking issue <https://github.com/rust-lang/rust/issues/120363>
   = note: this error originates in the attribute macro `savefile_abi_exportable` (in Nightly builds, run with -Z macro-backtrace for more info)
```

This happens in rust nightly from `rustc 1.83.0-nightly (1bc403daa 2024-10-11)` and newer.

**Comment by avl** [comment]
Reported to upstream rust ##131643

**Comment by avl** [comment]
This will be solved by #77
